### PR TITLE
Move from Hunter to VCPKG

### PR DESCRIPTION
**Pull request by Serafadam**
This PR introduces following changes:
- All dependencies from Hunter have been moved to VCPKG
- System dependencies (OpenCV, PCL) have also been moved to VCPKG
- VCPKG caching in GH Actions
- Manylinux image has been bumped to manylinux_2_28

### CODE REVIEW

File: .github/workflows/python-main.yml [hunk 1]
```diff
@@ -465,30 +478,37 @@ jobs:
...
       ACTIONS_ALLOW_USE_UNSECURE_NODE_VERSION: true
       DEPTHAI_VCPKG_CFLAGS: "-std=c99" # Needed so vpckg can bootstrap itself with the old GCC on the manylinux image
       VCPKG_FORCE_SYSTEM_BINARIES: "1" # Needed so vpckg can bootstrap itself
-      DEPTHAI_BUILD_BASALT: ON
-      DEPTHAI_BUILD_PCL: ON
-      DEPTHAI_BUILD_RTABMAP: ON
+      VCPKG_BINARY_SOURCES: "clear;x-gha,readwrite"
+      # DEPTHAI_BUILD_BASALT: ON
+      # DEPTHAI_BUILD_PCL: ON
+      # DEPTHAI_BUILD_RTABMAP: ON
```
Comment (moratom) [inline .github/workflows/python-main.yml:499]: Doesn't work?

Comment (Serafadam) [reply .github/workflows/python-main.yml]: Enabled now

**Review by moratom — COMMENTED** [review]
Thanks a lot!

Looking forward to get this in and say goodbye to Hunter :)



I left mostly nitpicks and questions, let's discuss on case by case basis what makes sense to address before merge and what we can split for later (for example moving as many packages as possible to stock)

File: .github/workflows/python-main.yml [hunk 2]
```diff
@@ -194,7 +204,10 @@ jobs:
         python-architecture: [x64] # TODO(Morato) - re-enable x86 - it complains that OpenCV even though it's 32 bit is not compatible
       fail-fast: false
     env:
-      DEPTHAI_BUILD_BASALT: ON
+      # DEPTHAI_BUILD_BASALT: ON
```
[the remark this answers is not in the document]
Comment (Serafadam) [reply .github/workflows/python-main.yml:207]: Must've commented out to speed up build, will reenable

File: .github/workflows/python-main.yml [hunk 3]
```diff
@@ -427,12 +443,6 @@ jobs:
       - name: Specify docstring to use while building the wheel
         run: echo "DEPTHAI_PYTHON_DOCSTRINGS_INPUT=$PWD/bindings/python/docstrings/depthai_python_docstring.hpp" >> $GITHUB_ENV
 
-      - name: Build and install depthai-core
-        run: |
-          cmake -S . -B build_core -D CMAKE_BUILD_TYPE=Release -D CMAKE_TOOLCHAIN_FILE=$PWD/cmake/toolchain/pic.cmake
-          cmake --build build_core --target install --parallel 4
-          echo "DEPTHAI_INSTALLATION_DIR=$PWD/build_core/install/" >> $GITHUB_ENV
-
       - name: Append build hash if not a tagged commit
```
[the remark this answers is not in the document]
Comment (Serafadam) [reply .github/workflows/python-main.yml:446]: Not sure if it was needed? Cache is now separate for each job

File: .github/workflows/test.workflow.yml [hunk 4]
```diff
@@ -72,6 +80,7 @@ jobs:
         export DISPLAY=:99
         xdpyinfo -display $DISPLAY >/dev/null 2>&1 || (Xvfb $DISPLAY &)
         source venv/bin/activate  # Activate virtual environment
+        python3 -m pip install jinja2
```
[the remark this answers is not in the document]
Comment (Serafadam) [reply .github/workflows/test.workflow.yml:83]: libsystemd which is a dependency of gtk

File: src/device/DeviceBase.cpp [hunk 5]
```diff
@@ -45,6 +45,7 @@
 #include "spdlog/spdlog.h"
 #include "utility/LogCollection.hpp"
 #include "utility/Logging.hpp"
+#include "utility/spdlog-fmt.hpp"
```
[the remark this answers is not in the document]
Comment (Serafadam) [reply src/device/DeviceBase.cpp:48]: Needed for spdlog with newer fmt versions

File: src/utility/spdlog-fmt.hpp [hunk 6]
```diff
@@ -15,7 +22,11 @@ struct fmt::formatter<dai::Path> : formatter<std::string> {
     // https://fmt.dev/latest/api.html#formatting-user-defined-types
     // https://fmt.dev/latest/syntax.html#format-specification-mini-language
     template <typename FormatContext>
-    auto format(const dai::Path& p, FormatContext& ctx) {
+#if FMT_VERSION >= 100000
+    auto format(const dai::Path& p, FormatContext& ctx) const {
+#else
```
[the remark this answers is not in the document]
Comment (Serafadam) [reply src/utility/spdlog-fmt.hpp:27]: https://github.com/luxonis/depthai-core/pull/1048

File: cmake/ports/g2o/portfile.cmake
[the remark this answers is not in the document]
Comment (Serafadam) [reply cmake/ports/g2o/portfile.cmake:1]: Unfortunately not, for now it needs our custom patch

File: cmake/ports/libarchive/portfile.cmake
[the remark this answers is not in the document]
Comment (Serafadam) [reply cmake/ports/libarchive/portfile.cmake:1]: Using stock causes issues with dai::Resources destructor, I haven't been able to pinpoint which part is exactly responsible for that

File: cmake/ports/rtabmap/portfile.cmake
[the remark this answers is not in the document]
Comment (Serafadam) [reply cmake/ports/rtabmap/portfile.cmake:1]: Current stock version has some issues, might get updated in the future

File: cmake/ports/tbb/portfile.cmake
[the remark this answers is not in the document]
Comment (Serafadam) [reply cmake/ports/tbb/portfile.cmake:1]: This specific version is required by basalt so until that codebase is update unfortunately not

File: cmake/ports/libnop/portfile.cmake
[the remark this answers is not in the document]
Comment (Serafadam) [reply cmake/ports/libnop/portfile.cmake:1]: Stock doesn't have CMake support sadly

File: CMakeLists.txt [hunk 7]
```diff
@@ -266,7 +258,7 @@ set(TARGET_OPENCV_NAME ${PROJECT_NAME}-opencv)
 set(TARGET_OPENCV_ALIAS opencv)
 if(DEPTHAI_OPENCV_SUPPORT)
     # Check if required libraries are available
-    set(REQUIRED_OPENCV_LIBRARIES "opencv_core" "opencv_imgproc" "opencv_videoio")
+    set(REQUIRED_OPENCV_LIBRARIES "opencv_core" "opencv_imgproc" "opencv_videoio" "opencv_highgui")
```
[the remark this answers is not in the document]
Comment (moratom) [reply CMakeLists.txt:272]: Hmm `zoo_hlper` shouldn't need OpenCV at all

File: cmake/ports/libnop/portfile.cmake
[the remark this answers is not in the document]
Comment (moratom) [reply cmake/ports/libnop/portfile.cmake:1]: It should still be found by `find_package` though right? We can iterate later

File: CMakeLists.txt [hunk 8]
```diff
@@ -496,6 +488,10 @@ set(CMAKE_OSX_DEPLOYMENT_TARGET 10.15)
 if(DEPTHAI_HAVE_OPENCV_SUPPORT AND DEPTHAI_MERGED_TARGET)
     # Link to OpenCV (publically)
     target_link_libraries(${TARGET_CORE_NAME} PUBLIC ${REQUIRED_OPENCV_LIBRARIES} ${THIRDPARTY_OPENCV_LIBRARIES})
+    # Link to FFMPEG (for video encoding)
+    target_include_directories(${TARGET_CORE_NAME} PRIVATE ${FFMPEG_INCLUDE_DIRS})
+    target_link_directories(${TARGET_CORE_NAME} PRIVATE ${FFMPEG_LIBRARY_DIRS})
+    target_link_libraries(${TARGET_CORE_NAME} PRIVATE ${FFMPEG_LIBRARIES})
```
[the remark this answers is not in the document]
Comment (Serafadam) [reply CMakeLists.txt]: Removed now

**Pull request merged**